core/src/eval/value/mod.rs

```diff
@@ -816,71 +1155,39 @@ impl NickelValue {
-                // Using `as_inline_unchecked` instead of `self.data` directly sidesteps the
-                // platform-specific differences
-                self.as_inline_unchecked() == other.as_inline_unchecked()
-            },
-        }
-    }
-
-    /// Try to update the position index of this value. This method will fail if `self` is a shared
-    /// value block; it won't re-allocate a new block automatically. It'll also fail if `self` is
-    /// an inline value but `pos_idx` is too large and can't be converted to a [InlinePosIdx].
-    pub fn with_pos_idx(self, pos_idx: impl Into<PosIdx>) -> Result<Self, PosIdxUpdateError> {
-        match self.tag() {
-            ValueTag::Pointer => {
-                // unwrap(): if the tag is `Pointer`, `ValueBlocRc::try_from(self)` must succeed.
-                let block: ValueBlockRc = self.try_into().unwrap();
-
-                if block.header().ref_count() == 1 {
-                    // Safety: if the ref count is equal to one, there's no sharing and we have
-                    // the exclusive mutable access over the block.
-                    unsafe { (*block.header_mut()).pos_idx = pos_idx.into() }
-                    Ok(block.into())
-                } else {
-                    Err(PosIdxUpdateError::SharedValueBlock { this: block.into() })
-                }
-            }
-            // Safety: the tag is checked to be `Inline`
             ValueTag::Inline => {
-                // [^irrefutable-let-patterns-32bits]: when compiling for 32-bit platforms, this
-                // pattern is irrefutable (the conversion is infallible), but we don't want to
-                // bother writing a separate version for 32-bit platforms.
-                #[allow(irrefutable_let_patterns)]
-                if let Ok(inline_pos_idx) = InlinePosIdx::try_from(pos_idx.into()) {
-                    Ok(self.with_inline_pos_idx(inline_pos_idx))
-                } else {
-                    Err(PosIdxUpdateError::NonInlinePosIdx { this: self })
-                }
+                other.is_inline()
+                    && unsafe {
+                        // Using `as_inline_unchecked` instead of `self.data` directly sidesteps the
+                        // platform-specific differences
+                        self.as_inline_unchecked() == other.as_inline_unchecked()
+                    }
             }
         }
     }
 
-    /// Same as [Self::with_pos_idx], but if `self` is a shared value block, returns a new copy
-    /// with the given position instead of failing. This method still fails if `self` is an inline
-    /// value and `pos_idx` is too large and can't be converted to an [InlinePosIdx].
-    pub fn force_with_pos_idx(self, pos_idx: impl Into<PosIdx>) -> Result<Self, Self> {
+    /// Updates the position of this value. First allocates a position index of the proper type
+    /// (inline or not) in the position table, and then use it as the new index.
```

Comment:
There is no longer a "proper type", right? All position indices are the same type now